core/src/main/scala/io/finch/Input.scala

```diff
@@ -32,6 +34,19 @@ final case class Input(request: Request, path: Seq[String]) {
   }
 
   /**
+   * Overrides (mutates) the payload of this input in order to reflect `a` as json and returns
+   * `this`.
+   *
+   * @note application/json will be passed as content type as well as the charset used to encode
+   *       the json
+   */
+  def withJson[A](a: A)(implicit e: Encode[A], cs: Option[Charset] = Some(Charsets.Utf8)): Input =
```

Comment:
It should be `Encode.Json[A]` so we provide compile-time guarantee on using JSON encoders here.